src/c++/perf_analyzer/genai-perf/tests/test_json_exporter.py

```diff
@@ -55,6 +83,64 @@ def test_generate_json(self, monkeypatch) -> None:
         config.artifact_dir = args.artifact_dir
         json_exporter = JsonExporter(config)
         assert json_exporter._stats_and_args == json.loads(self.expected_json_output)
+        json_exporter.export()
+        expected_filename = "profile_export_genai_perf.json"
+        written_data = [
+            data
+            for filename, data in mock_read_write
+            if os.path.basename(filename) == expected_filename
+        ]
+        if written_data == []:
+            raise Exception(
+                f"Expected file {expected_filename} not found in written data."
+            )
+        assert len(written_data) == 1
+        assert json.loads(written_data[0]) == json.loads(self.expected_json_output)
+
+    def test_generate_json_custom_export(
+        self, monkeypatch, mock_read_write: pytest.MonkeyPatch
+    ) -> None:
+        artifacts_dir = "artifacts/gpt2_vllm-triton-vllm-concurrency1"
+        custom_filename = "custom_export.json"
+        expected_filename = f"{artifacts_dir}/custom_export_genai_perf.json"
+        expected_profile_filename = f"{artifacts_dir}/custom_export.json"
+        cli_cmd = [
+            "genai-perf",
+            "profile",
+            "-m",
+            "gpt2_vllm",
+            "--backend",
+            "vllm",
+            "--streaming",
+            "--extra-inputs",
+            "max_tokens:256",
+            "--extra-inputs",
+            "ignore_eos:true",
+            "--profile-export-file",
+            custom_filename,
+        ]
+        monkeypatch.setattr("sys.argv", cli_cmd)
+        args, _ = parser.parse_args()
+        config = ExporterConfig()
+        config.stats = self.stats
+        config.args = args
+        config.extra_inputs = parser.get_extra_inputs_as_dict(args)
+        config.artifact_dir = args.artifact_dir
+        json_exporter = JsonExporter(config)
+        json_exporter.export()
+        written_data = [
+            data for filename, data in mock_read_write if filename == expected_filename
+        ]
```

Comment:
Same question as above.

Reply:
For JSONs, write is only called once. However, this is meant to be consistent with CSVs and allow for the case where write is called multiple times in case this changes in the future.